auth0/src/main/java/com/auth0/android/provider/OAuthManager.java

```diff
@@ -131,56 +125,21 @@ boolean resume(AuthorizeResult result) {
             return true;
         }
 
-        final Date expiresAt = !values.containsKey(KEY_EXPIRES_IN) ? null : new Date(getCurrentTimeInMillis() + Long.parseLong(values.get(KEY_EXPIRES_IN)) * 1000);
-        boolean frontChannelIdTokenExpected = parameters.containsKey(KEY_RESPONSE_TYPE) && parameters.get(KEY_RESPONSE_TYPE).contains(RESPONSE_TYPE_ID_TOKEN);
-        final Credentials frontChannelCredentials = new Credentials(frontChannelIdTokenExpected ? values.get(KEY_ID_TOKEN) : null, values.get(KEY_ACCESS_TOKEN), values.get(KEY_TOKEN_TYPE), null, expiresAt, values.get(KEY_SCOPE));
-
-        if (frontChannelIdTokenExpected) {
-            //Must be response_type=id_token (or additional values)
-            assertValidIdToken(frontChannelCredentials.getIdToken(), new VoidCallback() {
-                @Override
-                public void onSuccess(@Nullable Void ignored) {
-                    if (!shouldUsePKCE()) {
-                        //response_type=id_token or response_type=id_token token
-                        callback.onSuccess(frontChannelCredentials);
-                        return;
-                    }
-                    //response_type=id_token code
-                    pkce.getToken(values.get(KEY_CODE), new SimpleAuthCallback(callback) {
-
-                        @Override
-                        public void onSuccess(@NonNull Credentials credentials) {
-                            Credentials finalCredentials = mergeCredentials(frontChannelCredentials, credentials);
-                            callback.onSuccess(finalCredentials);
-                        }
-                    });
-                }
-
-                @Override
-                public void onFailure(@NonNull Auth0Exception error) {
-                    AuthenticationException wrappedError = new AuthenticationException(ERROR_VALUE_ID_TOKEN_VALIDATION_FAILED, error);
-                    callback.onFailure(wrappedError);
-                }
-            });
-            return true;
-        }
-
         if (!shouldUsePKCE()) {
```

Comment:
Do we need to keep a check for something that is documented to never happen? How can a developer recover the authentication from this unexpected scenario?

Reply:
Given it is documented to never happen, we could remove the check. That would mean if for some reason SHA_256 is not supported, and an exception is thrown when creating the code challenge as part of creating the `PKCE` instance, there would be a hard failure. But if it can never really happen, we can just allow it to propagate if for some reason it does happen.

Reply:
If we keep it, we should rename the method since there is no choice other than "using PKCE". 

The questions left are: what the developers can do when this error is raised? and how can they differentiate this error from any other error also raised by this sdk?